docs/design.md

```diff
@@ -104,3 +104,42 @@ several ways according to policy:
 
 Then like the usual LibOS pattern it will call the container's entry point as though the container
 was a process unto itself (rather than where it now sits like a library in the LibOS PAL process).
+
+## [Future Work] Local Attestation for Filesystem Decryption Key Provisioning
+
+![EnclaveCC_LA.png](images/EnclaveCC_LA.png)
+
+A local attestation (LA) feature is needed for authentication/authorization and secret provisioning
+between an Agent-enclave instance and a set of App-enclave instances resided in a same Pod. this
+design is a guide for enclave-CC developers to implement this feature.
+
+In this design, the authentication and authorization could be safely simplified in the context of
+enclave-CC by verifying and validating the SGX local report with customizable policy rules, e.g.,
+matching mrenclave/mrsigner combined with configid.
+
+In most scenarios, it is one-shot transactions used for single filesystem decryption key
+provisioning. However, this LA protocol can easily be extended to provision multiple keys for more
+general purposes.
+
+To achieve this, it is necessary to add a key id to the second request message sent along G_a and
+app's report, with the ID being used to indicate which key is being requested. But whether the key
+id can be automatically enumerated or not is decided by the caller ttrpc service/client and their
+contexts. At this moment, the caller is not discussed yet. We hope to merge the protocol definition
+completed prior to the implementation. By then, we could make the final decision to add the key id
+into protocol or not.
+
+In addition to adding this ID, the generated nonce and derived AEK also expect to be in the cache on
+both sides to keep the secure session alive which has been established in initial transaction. The
+session's security context will hold those cached values to ensure AIC triad and freshness for
+subsequent key provisioning requests.
+
+Furthermore, agent or app. should set the expiration time of its own security context properly to
+minimize risk exposure. Upon closing of underlying ttrpc connection, the security context will be
+cleared accordingly.
+
+Please note that there is no secure transport configured between ttrpc endpoints in our case,
+therefore the LibOSes need to validate and check integrity of whatever they received. The ttrpc
+service needs to be strictly configured to listen on loopback interface only for security. With
+regard to the secret in transit, it was designed to be wrapped by a shared key. In addition, the
+identity of agent or app. could also be verified to further ensure the provenance of the app.
+enclave container.
```

Comment:
In the diagram please consistently name the FUSE key. Sometimes it is just mentioned as `key` which can be ambiguous in a key exchange protocol.

can you improve the wording in this step...
<img width="295" alt="image" src="https://user-images.githubusercontent.com/4999494/187075179-bf7bee86-90a9-4551-9919-5b38373d4884.png">

Something like `Encrypt` the fuse key using the AEK rather than `wrap`. 
Add Nonce 1 in `Associated Data`

Reply:
rephrased it and other associated parts, thanks.